src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -689,14 +691,14 @@ public static T Find<T>(T[] array, Predicate<T> match)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.match);
             }
 
-            for (int i = 0; i < array!.Length; i++) // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/538
+            for (int i = 0; i < array!.Length; i++) // TODO-NULLABLE: Remove ! when [DoesNotReturn] respected
             {
-                if (match!(array[i])) // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/538
+                if (match!(array[i])) // TODO-NULLABLE: Remove ! when [DoesNotReturn] respected
                 {
                     return array[i];
                 }
             }
-            return default!; // TODO-NULLABLE-GENERIC
+            return default!;
```

Comment:
Why will this never be able to be done without a !? Shouldn't the compiler see that the return type `MaybeNull`?

Reply:
Not never, but for C# 8.  My understanding of the plan, given limited time before the release, is that the attributes won't affect inwardly in a method.

Reply:
Ok, cool.